app/platform/fabric/sync/SyncService.ts

```diff
@@ -399,6 +423,24 @@ export class SyncServices {
 		} else {
 			logger.debug('Missing blocks not found for %s', channel_name);
 		}
+		if (this.validatePurgemode(purgeMode)) {
+			let response: boolean;
+			switch (purgeMode) {
+				case purge_mode[0]: {
+					response = await this.persistence.
+						getCrudService().
+						deleteOldData(channel_genesis_hash, network_id, daysToPurge);
+					break;
+				}
+				case purge_mode[1]: {
```

Comment:
Can you check if we can use the enums here instead of `purge_mode[0] and purge_mode[1]`